pennylane/templates/subroutines/out_adder.py

```diff
@@ -0,0 +1,134 @@
+
+    We can see that the result [1 0 0] corresponds to 4, which comes from :math:`5+6=11 \longrightarrow 11 \, \text{mod} \, 7 = 4`.
+    """
+
+    grad_method = None
+
+    def __init__(self, x_wires, y_wires, output_wires, mod=None, work_wires=None, id=None):
+
+        if mod is None:
+            mod = 2 ** (len(output_wires))
+        if (not hasattr(output_wires, "__len__")) or (mod > 2 ** len(output_wires)):
+            raise ValueError("OutAdder must have at least enough wires to represent mod.")
+        if work_wires is not None:
+            if any(wire in work_wires for wire in x_wires):
+                raise ValueError("None of the wires in work_wires should be included in x_wires.")
+            if any(wire in work_wires for wire in y_wires):
+                raise ValueError("None of the wires in work_wires should be included in y_wires.")
+        else:
+            max_wire = max(max(x_wires), max(y_wires), max(output_wires))
+            work_wires = [max_wire + 1, max_wire + 2]
+        for key in ["x_wires", "y_wires", "output_wires", "work_wires"]:
+            self.hyperparameters[key] = qml.wires.Wires(locals()[key])
+        all_wires = sum(
+            self.hyperparameters[key]
+            for key in ["x_wires", "y_wires", "output_wires", "work_wires"]
+        )
+        self.hyperparameters["mod"] = mod
+        super().__init__(wires=all_wires, id=id)
+
+    @property
+    def num_params(self):
+        return 0
+
+    @staticmethod
+    def compute_decomposition(x_wires, y_wires, output_wires, mod, work_wires, **kwargs):
+        r"""Representation of the operator as a product of other operators.
+        Args:
+            x_wires (Sequence[int]): the wires that stores the integer :math:`x`.
+            y_wires (Sequence[int]): the wires that stores the integer :math:`y`.
+            output_wires (Sequence[int]): the wires that stores the sum modulo mod :math:`x+y mod mod`.
+            mod (int): modulo with respect to which the sum is performed, default value will be ``2^len(output_wires)``.
+            work_wires (Sequence[int]): the auxiliary wires to use for the sum modulo :math:`mod` when :math:`mod \neq 2^{\textrm{len(output_wires)}}`.
+        Returns:
+            list[.Operator]: Decomposition of the operator
+
+        **Example**
+
+        >>> qml.OutAdder.compute_decomposition(x_wires=[0,1], y_wires=[2,3], output_wires=[5,6], mod=4, work_wires=[4,7])
+        [CNOT(wires=[2, 5]),
+        CNOT(wires=[3, 6]),
+        QFT(wires=[5, 6]),
+        ControlledSequence(PhaseAdder(wires=[5, 6]), control=[0, 1]),
+        Adjoint(QFT(wires=[5, 6]))]
+        """
+        op_list = []
+        if mod != 2 ** (len(output_wires)):
+            qft_new_output_wires = work_wires[:1] + output_wires
+        else:
+            qft_new_output_wires = output_wires
+        for i in range(len(y_wires)):
+            op_list.append(qml.CNOT(wires=[y_wires[i], output_wires[i]]))
```

Comment:
It is well thought out but better not to do it that way. This way the adder will only work if the initial input is zero. What we are looking for is to accumulate the previous result. We would have to do the same as you do later: ControlledSequence of the Adder(1)

Reply:
That's true, thanks for noticing! Now I fixed it removing the CNOTs and adding an extra ControlledSequence of the PhaseAdder (1) controlled by y_wires. I also included some tests as you suggested and it works locally 🚀